geco/graphs/utilities.py

```diff
@@ -35,6 +35,73 @@ def edgeweight_properties(graph):
     return max_weight, min_weight, num_of_zero_weights
 
 
+def num_of_simple_cycles(graph):
+    """
+    Finds number of simple cycles
+
+    Parameters
+    ----------
+    graph: nx.Graph
+        Graph to find number of simple cycles in
+
+    Returns
+    -------
+    num_of_simple_cycles: int
+        Number of simple cycles in graph
+
+    Notes
+    -----
+    It was pointed out that this implementation
+    is probably buggy in the undirected case.
+    """
+    if graph.is_directed():
+        return len(list(nx.simple_cycles(graph)))
```

Comment:
Small python trick:
Instead of collecting the generator into a list and returning its length, one can avoid allocating memory by just
`return sum(1 for _ in nx.simple_cycles(graph))`